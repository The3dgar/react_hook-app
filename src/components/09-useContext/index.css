body {
  padding: 0px;
  background-color: white;
  color: black
}