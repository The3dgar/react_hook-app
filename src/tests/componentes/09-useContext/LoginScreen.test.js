import React from "react";
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";
import { shallow, mount } from "enzyme";
import { demoUser } from "../../fixture/demoUser";
import { UserContext } from "../../../components/09-useContext/UserContext";
import "@testing-library/jest-dom";

describe("Pruebas en <LoginScreen/>", () => {
  const setUser = jest.fn();
  // wrap
  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test("should displayed correctly", () => {
    // snap
    expect(wrapper).toMatchSnapshot();
  });
  test("should execute setUser", () => {
    wrapper.find("button").simulate("click");

    expect(setUser).toHaveBeenCalledWith({ id: 123, name: "edgar" });
  });
});
