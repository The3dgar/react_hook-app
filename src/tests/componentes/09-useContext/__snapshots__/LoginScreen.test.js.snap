// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> should displayed correctly 1`] = `
<div>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Change User
  </button>
</div>
`;
